libdnf/rpm/reldep_list.cpp

```diff
@@ -95,8 +95,23 @@ bool ReldepList::add_reldep_with_glob(const std::string & reldep_str) {
     Dataiterator di;
     dataiterator_init(&di, pool, 0, 0, 0, dep_splitter.get_name_cstr(), SEARCH_STRING | SEARCH_GLOB);
     while (dataiterator_step(&di)) {
-        ReldepId id = Reldep::get_reldep_id(sack, di.kv.str, dep_splitter.get_evr_cstr(), dep_splitter.get_cmp_type());
-        add(id);
+        switch (di.key->name) {
+            case SOLVABLE_PROVIDES:
```

Comment:
And what about SOLVABLE_NAME?

Reply:
Is it not added to provides automatically?

Reply:
No. Package libdnf.s390 do not provide anything on x86_64 system. Same for src packages.
The other question would be, is it a problem? We suggest that such a reldep will be used for queries therefore anyway you will be not able to match anything.

Reply:
Right. I don't know, the way I understand it, this method is quite special. It depends on the intended semantics, its docstring is not very clear, it should say something like:
> Matches `reldep_str` against all reldep data in the pool and adds all matching reldeps to the list.

The method is on the interface, even though it's internally used in queries doesn't mean someone can't find an inventive use outside that.

I think the special nature of the method warrants it rather shouldn't be an "add" method of the ReldepList, I think it should be a method on perhaps the Sack (maybe elsewhere?) like this:
```
ReldepList Sack::get_all_reldeps_matching_glob(const std::string & glob);
```

The reasoning is the ReldepList should primarily be a container with simple methods to manipulate the contents, clever logic like the one being done here should be somewhere up the class hierarchy.

That said, I'm still not sure about the name. I guess it is a reldep identifier and so should be part of the list?

Reply:
I suggest that we can do with the present implementation.

Reply:
We've agreed to leave out SOLVABLE_NAME for now, so this should be good as it is.